### ISSUE TITLE
Wiki

### ISSUE THREAD

**Issue by ProbablePrime** [Yellow-Dog-Man/Resonite-Issues#1]
**Is your feature request related to a problem? Please describe.**
A lack of reliable information at a permanent link

**Describe the solution you'd like**
A publicly available Wiki that documents frequently asked questions and other information.

**Additional context**
I'm making this to pre-bake it for when this repo goes public.

**Comment by FlipperLP** [comment]
Do you already have a wiki-provider/-solution in mind, and should it be editable by the community or just read-only?

**Comment by ProbablePrime** [comment]
- We have a solution in mind, It is up and running and working
- It will be editable by everyone.

**Comment by FlipperLP** [comment]
What software do you guys have in mind, and when will be the go-live date?

**Comment by ProbablePrime** [comment]
Media Wiki and when it is ready.

**Comment by FlipperLP** [comment]
Seems kinda dated, but alright.

**Comment by ProbablePrime** [comment]
Mediawiki is very much alive and gets updates everyday. You may be referring to its look and feel which we will be messing with, most likely after launch, So expect the default theme to start.

**Comment by ProbablePrime** [comment]
Other priorities keep trumping this one unfortunately(https://github.com/Yellow-Dog-Man/Resonite-Issues/blob/main/HOW_WE_PRIORITIZE.md)

My soul burns every day this is not done. I'll keep ya'll up to date.

**Comment by clarjon1** [comment]
Is there anything the community can do to help pitch in on this? Eg harvesting and collating info, updating Old Platform docs with new verbiage, digging up / creating useful flexible theme(s)?

**Comment by shininghero** [comment]
I second clarjon's post.  Don't be afraid to let us help.
Give us a style guide or sample document, and we can at least offer to fill out some of the component or flux node docs as we go.

**Comment by ProbablePrime** [comment]
There is also not a content gap, well content isn't the main focus. So I'm not entirely sure what can be done to help until this launches.

After launch of course, we'll need more content though.

**Comment by FlipperLP** [comment]
I think the most important part would be to actually have something running and guidelines what content goes where, as well as how a Wiki Page should look like/what information should always be available.
So maaaaybe a few articles would be good to have an Idea what you have in mind.

---

If it wouldn't be already decided on MediaWiki I would suggest a solution based on Git(Hub) and Markdown. That way we don't lose PRs and a unified place to manage code/data on a file level. While I observed an edit history in MediaWiki, I personally prefer the Git way with Markdown. Review as a PR _before_ it gets published. While I understand that it is more work on your end approving PRs, it's a better way to proofread, propose edits and others chip in to improve. ~~On the other hand, monitoring every edit so a funni doesn't happen sounds like more work to me, especially when it is already published/live.~~ EDIT: There is a current and preview feature, but don't know about Discussion.
I am also unsure on possible migration paths away from MediaWiki, in case things turn sour, if it is not just a bunch of files.
But again just a thought. (I never really worked with MediaWiki, so don't quote me on that stuff above, and I would be interested, if MediaWiki offers something similar or if the file structure can be edited with Markdown/HTML/whatever.)
[Here are some projects](https://github.com/testthedocs/awesome-docs#site-generators) (and many more I can't find rn), that use Git to generate the Wiki. The only real issue I can see happening, is when it comes to pictures and how that is handled and that people don't necessarily know how to use Git.
While I respect the decision and in the end we need to move forward to whatever, so we can show progress. But I felt like I need to explain my previous comment further and offer some alternatives to where the Wiki/Documentation is heading towards.

**Comment by Banane9** [comment]
Thought I just had while reading #937: When the wiki (in whatever shape) is actually there, it would be great if every component / protoflux node / tool / window could have a direct link to the relevant page in one way or another. This could be a typical (i) icon, with some recognizable Resonite styling, of course - either directly on the UI for windows and components, or as a context menu option for tools. Protoflux nodes may get too cluttered with an extra element, so for those it could also be a context menu entry that opens the wiki page for the node type that would currently be spawned.

Anyways, the reason I'm mentioning it is, that this would become very easy to implement, if the pages just used the (full) name of the component type for their url, so generating the correct link would be trivial to do.

**Comment by Merith-TK** [comment]
The 

> I think the most important part would be to actually have something running and guidelines what content goes where, as well as how a Wiki Page should look like/what information should always be available. So maaaaybe a few articles would be good to have an Idea what you have in mind.
> 
> If it wouldn't be already decided on MediaWiki I would suggest a solution based on Git(Hub) and Markdown. That way we don't lose PRs and a unified place to manage code/data on a file level. While I observed an edit history in MediaWiki, I personally prefer the Git way with Markdown. Review as a PR _before_ it gets published. While I understand that it is more work on your end approving PRs, it's a better way to proofread, propose edits and others chip in to improve. ~On the other hand, monitoring every edit so a funni doesn't happen sounds like more work to me, especially when it is already published/live.~ EDIT: There is a current and preview feature, but don't know about Discussion. I am also unsure on possible migration paths away from MediaWiki, in case things turn sour, if it is not just a bunch of files. But again just a thought. (I never really worked with MediaWiki, so don't quote me on that stuff above, and I would be interested, if MediaWiki offers something similar or if the file structure can be edited with Markdown/HTML/whatever.) [Here are some projects](https://github.com/testthedocs/awesome-docs#site-generators) (and many more I can't find rn), that use Git to generate the Wiki. The only real issue I can see happening, is when it comes to pictures and how that is handled and that people don't necessarily know how to use Git. While I respect the decision and in the end we need to move forward to whatever, so we can show progress. But I felt like I need to explain my previous comment further and offer some alternatives to where the Wiki/Documentation is heading towards.

While I agree with this, github markdown is not without its drawbacks. namely extra steps required in order to update the wiki

alot of the time these pages use what are called "Static Site Generators", they take markdown (or other markup languages) and dynamically "compile" the markdown into user readable HTML. Which requires not only pulling the latest changes, but also pushing the updated website, 

Media Wiki is, mostly automated outside of maybe adding an review setting

**Comment by Banane9** [comment]
> alot of the time these pages use what are called "Static Site Generators", they take markdown (or other markup languages) and dynamically "compile" the markdown into user readable HTML. Which requires not only pulling the latest changes, but also pushing the updated website,

That can easily be automated with actions and could (theoretically) even be hosted on GitHub directly using pages. The prime benefit is also, that the information isn't siloed away into some platform but available as text that could directly be read and edited if necessary.
See this interesting video on the topic: https://www.youtube.com/watch?v=WgV6M1LyfNY

**Comment by FlipperLP** [comment]
Manual deployment would be way too much work. So I wouldn't have set up anything else other than GitHub Actions.
Amazing Video, I have to watch them more.

**Comment by ProbablePrime** [comment]
Lets focus on the current solution here rather than discussing the intricacies of static site generators please.

**Comment by FlipperLP** [comment]
Marry late Christmas.
So you are definitely going to settle for MediaWiki, or what do you mean with “current solution”?

**Comment by clarjon1** [comment]
As shown below, Prime already answered about it being mediawiki in response to your query on oct 31. The `current solution` is **MediaWiki**, which is running in a staging environment currently only accessible to the YDMS team, which will be made publicly available when they feel it's at a state releasable to the public.
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/627845/d4a9e366-3fe8-4fd6-9d53-1deccae169d6)

**Comment by FlipperLP** [comment]
Alright

**Comment by Lexevolution** [comment]
There now seems to be no thorough FrooxEngine related dcumentation on any sites (be it first or third party). What could we as a community do to make the wiki higher priority?

**Comment by Frooxius** [comment]
I don't think there is super much that can be done from community side right now, because it's already pretty high priority on our end, but it keeps getting pushed back by other higher priority items, that we needed to be tackled in time (for example GDPR related work, resolving server/relay issues and so on).

It's mostly just work-bandwidth issue on our end, with a bunch of post-launch problems and issues that need to be sorted out. I know it sucks, but it'll get sorted out in time.

**Comment by ProbablePrime** [comment]
I have an update, 

We should be ready by the 15th, likely much before that but hey here's a date.

**Comment by ProbablePrime** [comment]
Does anyone have any requested mediawiki plugins? I'm going through plugin operations today for some security things.

**Comment by JackTheFoxOtter** [comment]
> Does anyone have any requested mediawiki plugins? I'm going through plugin operations today for some security things.

The one that allows you to access PDF versions of wiki pages through a direct URL, now that we have native PDF viewing ability.

**Comment by DoubleStyx** [comment]
> Does anyone have any requested mediawiki plugins? I'm going through plugin operations today for some security things.

Maybe the Math extension, for rendering LaTeX in-line?

**Comment by TisFoolish** [comment]
Maybe the Discord extension with a channel to follow wiki edits?

https://m.mediawiki.org/wiki/Extension:Discord

**Comment by lxw404** [comment]
Mermaid allows you to render graphs/charts/UML diagrams with minimal syntax which could be useful:

[https://www.mediawiki.org/wiki/Extension:Mermaid](https://www.mediawiki.org/wiki/Extension:Mermaid)

**Comment by Lexevolution** [comment]
Here's a few that are [bundled with MediaWiki](https://www.mediawiki.org/wiki/Bundled_extensions_and_skins) that I'd like to see enabled:
- [Cite](https://www.mediawiki.org/wiki/Extension:Cite) (Allows footnote citations)
- [Math](https://www.mediawiki.org/wiki/Extension:Math) (Allows math notation)
- [Syntax Highlight](https://www.mediawiki.org/wiki/Extension:SyntaxHighlight) (Colours code blocks to show programming language syntax)
- [VisualEditor](https://www.mediawiki.org/wiki/Extension:VisualEditor) (Allows a much easier experience editing a wiki page, using a WYSIWYG style editor)
- [WikiEditor](https://www.mediawiki.org/wiki/Special:MyLanguage/Extension:WikiEditor) (Adds some features to the default wiki editor to allow for easier formatting)
- [CategoryTree](https://www.mediawiki.org/wiki/Extension:CategoryTree) (Adds a tree view for wiki page categories. Might be helpful in the future)

For @JackTheFoxOtter's suggestion, [ElectronPdfService](https://www.mediawiki.org/wiki/Extension:ElectronPdfService) is the one used and maintained by Wikimedia, and would provide the same functionality as Wikipedia's version.

[MobileFrontend](https://www.mediawiki.org/wiki/Extension:MobileFrontend) seems good, to make wiki pages easier to read on mobile devices
[AdvancedSearch](https://www.mediawiki.org/wiki/Extension:AdvancedSearch) could be good for the complex nature of Resonite

**Comment by ProbablePrime** [comment]
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/8791132/dd5981fe-6cd6-409d-8ebf-b47b1d60b7b8)

**Comment by ProbablePrime** [comment]
PDF is going to have to wait, it is a complex extension. I don't know why it needs to be though :(

**Comment by ProbablePrime** [comment]
Ok we got PDF's up and running, they're a bit ugly and images dont work right now but we'll work on that after launch.
I used: https://www.mediawiki.org/wiki/Extension:Mpdf instead, which also required some updates due to our PHP version which will go in: https://github.com/Yellow-Dog-Man/mediawiki-extensions-Mpdf

At some point i'll make images work, but very complicated. The other PDF extension requires a REST server AND an Electron Renderer server to be created and maintained, so Mpdf sounded better.
